File: kinesis-data-streams-java/src/test/java/com/github/antontucek/kinesis/data/streams/java/OrderTest.java
package com.github.antontucek.kinesis.data.streams.java;

import org.junit.jupiter.api.Assertions;
import java.time.Instant;
import java.time.LocalDateTime;
import java.time.ZoneId;
import java.util.List;


class OrderTest {

    @org.junit.jupiter.api.Test
    void fromToJsonAsBytes() {
        Instant instant = LocalDateTime.of(2001, 1, 1, 0, 0).atZone(ZoneId.of("UTC")).toInstant();
        OrderItem orderItem = new OrderItem("Jet", 1000.0F, 1);
        Order order1 = new Order(instant, "CZ", "Jan", List.of(orderItem));
        Order order2 = Order.fromJsonAsBytes(order1.toJsonAsBytes());
        Assertions.assertEquals(order1, order2);
    }
}

File: kinesis-data-streams-kotlin/src/test/kotlin/com/github/antontucek/kinesis/data/streams/kotlin/OrderTest.kt
package com.github.antontucek.kinesis.data.streams.kotlin

import com.fasterxml.jackson.module.kotlin.readValue
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import java.time.LocalDateTime
import java.time.ZoneId

class OrderTest {

    @Test
    fun fromToJsonAsBytes() {
        val instant = LocalDateTime.of(2001, 1, 1, 0, 0).atZone(ZoneId.of("UTC")).toInstant()
        val orderItem = OrderItem("Jet", 1000.0F, 1)
        val order1 = Order(instant, "CZ", "Jan", listOf(orderItem))
        val order2: Order = Order.mapper.readValue(Order.mapper.writeValueAsBytes(order1))
        assertThat(order1).isEqualTo(order2)
    }
}